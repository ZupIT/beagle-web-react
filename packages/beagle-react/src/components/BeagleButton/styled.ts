/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { BeagleTheme } from '../commons.styled'

export const StyledBeagleButton = styled.button`
  border: 1px solid ${BeagleTheme.swampLight};
  line-height: 34px;
  color: inherit;
  background: transparent;
  box-sizing: border-box;
  position: relative;
  cursor: pointer;
  outline: none;
  display: inline-block;
  white-space: nowrap;
  text-decoration: none;
  vertical-align: baseline;
  text-align: center;
  margin: 0;
  min-width: 70px;
  padding: 0 16px;
  border-radius: 5px;
  overflow: visible;
  
  &:hover {
    background-color: ${BeagleTheme.swampTransparent};
  }

  &:disabled {
    opacity: 0.4;
    pointer-events: none;
  }
`
