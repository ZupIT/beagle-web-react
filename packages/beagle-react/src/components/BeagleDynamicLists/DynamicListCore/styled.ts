/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { Direction } from 'common/models'

interface StyledDynamicViewsInterface {
  direction: Direction,
  useParentScroll?: boolean,
  isScrollIndicatorVisible?: boolean,
  listType?: string,
  spanCount?: number,
}

export const StyledDynamicViewsInterface = styled.div<StyledDynamicViewsInterface>`
  display: ${({ listType }) => listType === 'GRID' ? 'grid' : 'flex'};
  grid-auto-flow: ${({ direction }) => direction === 'HORIZONTAL' ? 'column' : 'row'};
  grid-template-columns: ${({ listType, spanCount, direction }) =>
    (listType === 'GRID' && spanCount && direction === 'VERTICAL') && `repeat(${spanCount}, 1fr)`};
  grid-template-rows: ${({ listType, spanCount, direction }) =>
    (listType === 'GRID' && spanCount && direction === 'HORIZONTAL')
    && `repeat(${spanCount}, 1fr)`};
  justify-content: ${({ listType, direction }) => listType === 'GRID'
    && direction === 'VERTICAL' && 'space-evenly'};
  flex-direction: ${({ direction }) => direction === 'VERTICAL' ? 'column' : 'row'};
  overflow: ${({ useParentScroll }) => useParentScroll ? 'inherit' : 'auto'};
  width: ${({ direction }) => direction === 'HORIZONTAL' ? '100%' : 'auto'};
  height: ${({ direction }) => direction === 'VERTICAL' ? '100%' : 'auto'};
  & ::-webkit-scrollbar {
    display: ${({ isScrollIndicatorVisible }) => isScrollIndicatorVisible ? 'auto' : 'none'};
  }
  -ms-overflow-style: ${({ isScrollIndicatorVisible }) => isScrollIndicatorVisible ?
    'auto' : 'none'};
  scrollbar-width: ${({ isScrollIndicatorVisible }) => isScrollIndicatorVisible ? 'auto' : 'none'};
`
