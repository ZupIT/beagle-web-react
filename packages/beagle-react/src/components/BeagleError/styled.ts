/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import BeagleButton from '../BeagleButton'

export const StyledRetryContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  margin: 2em;
  font-size: 23px;
`

export const StyledRetryButton = styled(BeagleButton)`
  max-width: 200px;
  margin: 8px 4px 4px 4px;
`

export const StyledShowMoreButton = styled(BeagleButton)`
  max-width: 200px;
  margin: 4px;
  border: none;
`

export const ErrorsDetailedContainer = styled.section`
  width: 100%;
  max-width: 800px;
  max-height: 0;
  opacity: 0;
  overflow: auto;
  position: relative;
  margin: 4px;
  border-radius: 8px;
  border: 1px solid #FFD7D7;
  background-color: #FFF0F0;
  padding: 8px;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  gap: 8px;
  transition: all 0.2s ease;

  &.show {
    max-height: 1024px;
    opacity: 1;
  }

  .error-message {
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    position: relative;
    color: #FF0000;
    font-size: 14px;
    padding: 0;
    margin: 0;

    .error-indicator {
        color: #414a4c;
        margin: 0 4px 0 0;
        font-size: 12px;
    }
  }
`
