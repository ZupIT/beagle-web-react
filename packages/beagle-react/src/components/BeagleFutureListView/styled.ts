/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { Direction } from 'common/models'

interface StyledListViewInterface {
  direction: Direction,
  useParentScroll?: boolean,
}

export const StyledListView = styled.div<StyledListViewInterface>`
  display: flex;
  flex-direction: ${({ direction }) => direction === 'VERTICAL' ? 'column' : 'row'};
  overflow: ${({ useParentScroll }) => useParentScroll ? 'inherit' : 'auto'};
  width: ${({ direction }) => direction === 'HORIZONTAL' ? '100%' : 'auto'};
  height: ${({ direction }) => direction === 'VERTICAL' ? '100%' : 'auto'};
`
