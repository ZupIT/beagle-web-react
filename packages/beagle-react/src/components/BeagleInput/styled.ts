/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'

interface StyledTextInputInterface {
  error?: string,
  showError?: boolean,
}

export const errorColor = '#FF0000'

export const StyledInput = styled.input<StyledTextInputInterface>`
  border: ${({ error, showError }) => error && showError ? errorColor : ''};
  outline: ${({ error, showError }) => error && showError ? `auto ${errorColor}` : ''};
`
