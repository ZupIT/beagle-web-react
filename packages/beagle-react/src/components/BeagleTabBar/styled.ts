/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import styled from 'styled-components'
import { BeagleTheme } from '../commons.styled'
import BeagleImage from '../BeagleImage'

export const StyledTabBar = styled.div`
  display: flex;
  align-items: center;
`

export const StyledBeagleTabItem = styled.div`
  display: flex;
  flex-direction: column;
  margin: 0 10px;
`

export const StyledBeagleTabItemContent = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  min-width: 20px;
  min-height: 15px;

  &:hover {
    background-color: ${BeagleTheme.swampTransparent};
  }
`

export const StyledBeagleImage= styled(BeagleImage) `
  max-width: 80px;
  max-height: 80px;
`

export const StyledSelected = styled.div`
  height: 2px;
  background-color: ${BeagleTheme.swamp};
`
