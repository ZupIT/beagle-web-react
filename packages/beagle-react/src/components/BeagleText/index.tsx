/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

import React, { FC } from 'react'
import { BeagleTextInterface } from 'common/models'
import { buildAccessibility } from '../../../../common/utils/accessibility'
import withTheme from '../utils/withTheme'
import { StyledText } from './styled'

const BeagleText: FC<BeagleTextInterface> = props => {
  const { text, className, textColor, alignment, style, accessibility } = props
  const a11y = buildAccessibility(accessibility)

  return (
    <StyledText 
      textColor={textColor} 
      alignment={alignment} 
      className={className} 
      style={style}
      {...a11y}>
      {text}
    </StyledText>
  )
}

export default withTheme(BeagleText)
