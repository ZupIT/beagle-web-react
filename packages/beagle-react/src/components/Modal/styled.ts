import styled from 'styled-components'
import { BeagleTheme } from '../commons.styled'

export const StyledModal = styled.div`
  background: ${BeagleTheme.darkGray};
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  z-index: 1;
  display: flex;
  align-items: center;
  justify-content: center;

  > div {
    background-color: #FFF;
    border-radius: 25px;
    padding: 30px;
  }
`
