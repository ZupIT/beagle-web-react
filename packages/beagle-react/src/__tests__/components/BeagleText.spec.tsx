/*
  * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
  *  http://www.apache.org/licenses/LICENSE-2.0
  *
  * Unless required by applicable law or agreed to in writing, software
  * distributed under the License is distributed on an "AS IS" BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
*/

// Link.react.test.js
import React from 'react'
import 'jest-styled-components'
import Adapter from 'enzyme-adapter-react-16'
import { configure, mount, ReactWrapper } from 'enzyme'
import BeagleText from '../../components/BeagleText'
import { StyledText } from '../../components/BeagleText/styled'

const initialText = 'Test'

configure({ adapter: new Adapter() })

test('Beagle snapshot Text', () => {
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={initialText} className='text-class'/>)
  expect(wrapper).toMatchSnapshot()
})

test('it should render the string as initialized', () => {
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={initialText} />)
  const p = wrapper.find(StyledText).getDOMNode<HTMLParagraphElement>()
  expect(p?.innerHTML).toBe(initialText)
})

test('it should render not render any text when text is null', () => {
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={null} />)
  const p = wrapper.find(StyledText).getDOMNode<HTMLParagraphElement>()
  expect(p?.innerHTML.length).toBe(0)
})

test('it should render not render any text when text is undefined', () => {
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={undefined} />)
  const p = wrapper.find(StyledText).getDOMNode<HTMLParagraphElement>()
  expect(p?.innerHTML.length).toBe(0)
})

test('it should render numbers as strings', () => {
  const value = 123.45
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={value} />)
  const p = wrapper.find(StyledText).getDOMNode<HTMLParagraphElement>()
  expect(p?.innerHTML).toBe(String(value))
})

test('it should render an object as stringfied json object', () => {
  const obj = { my: 'test' }
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={obj} />)
  const p = wrapper.find(StyledText).getDOMNode<HTMLParagraphElement>()
  expect(p?.innerHTML).toBe(JSON.stringify(obj))
})

test('it should render an array as stringfied json array', () => {
  const arr = [{ my: 'test' }, 'test', 123.45]
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={arr} />)
  const p = wrapper.find(StyledText).getDOMNode<HTMLParagraphElement>()
  expect(p?.innerHTML).toBe(JSON.stringify(arr))
})

test('it should render an function as empty string', () => {
  const fun = function test() { return true }
  const wrapper: ReactWrapper<typeof BeagleText> = mount(<BeagleText text={fun} />)
  const p = wrapper.find(StyledText).getDOMNode<HTMLParagraphElement>()
  expect(p?.innerHTML).toBe('')
})
