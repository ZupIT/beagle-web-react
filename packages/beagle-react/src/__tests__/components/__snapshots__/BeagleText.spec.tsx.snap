// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Beagle snapshot Text 1`] = `
<BeagleText
  className="Test Class"
  text="Test"
/>
`;
